src/ansi-c/c_typecheck_type.cpp

```diff
@@ -820,12 +820,24 @@ void c_typecheck_baset::typecheck_compound_type(struct_union_typet &type)
     }
   }
 
-  symbol_typet symbol_type(identifier);
-  symbol_type.add_source_location()=type.source_location();
+  if(type.id() == ID_union)
+  {
+    union_tag_typet tag_type(identifier);
+    tag_type.add_source_location() = type.source_location();
+
+    c_qualifierst original_qualifiers(type);
```

Comment:
This seems common to both branches and could be done beforehand.

Reply:
done